src/post.js

```diff
@@ -10,12 +10,42 @@ module.exports = async config => {
   const name = `v${pkg.version}`;
   const tag = {owner, repo, ref: `refs/tags/${name}`, sha: await promisify(gitHead)()};
   const body = await promisify(plugins.generateNotes)(config);
-  const release = {owner, repo, tag_name: name, name, target_commitish: branch, draft: !!debug, body};
+  let githubPreRelease = false;
+  let release = {
+    owner,
+    repo,
+    tag_name: name,
+    name,
+    target_commitish: branch,
+    draft: !!debug,
+    body,
+    prerelease: githubPreRelease,
+  };
 
   if (debug && !githubToken) {
     return {published: false, release};
   }
 
+  // obtain npm dist tag if any
+  let npmDistTag = (pkg.publishConfig || {}).tag;
+
+  // if npmDistTag is set, we can consider github pre release
+  if (npmDistTag != null) {
+    // obtain release.no-pre-release config
+    const noPreReleaseTags = (pkg.release || {})['no-pre-release'] || ['latest'];
+
+    // make sure no-pre-release isn't the same as npm manually set dist tag,
+    // nor includes it when set as an Array
+    if (
+      (typeof noPreReleaseTags === 'string' && noPreReleaseTags !== npmDistTag) ||
+      (Array.isArray(noPreReleaseTags) && noPreReleaseTags.indexOf(npmDistTag) === -1)
```

Comment:
You can use [includes()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/includes) here.